examples/ipex/deepspeed/run_generation_deepspeed.py

```diff
@@ -0,0 +1,154 @@
+from optimum.intel import IPEXModelForCausalLM
+
+
+sys.path.append(sys.path[0] + "/../../")
+logger = logging.getLogger(__name__)
+
+
+torch._C._jit_set_texpr_fuser_enabled(False)
+try:
+    ipex._C.disable_jit_linear_repack()
+except Exception:
+    pass
+
+
+def get_int_from_env(env_keys, default):
+    """Returns the first positive env value found in the `env_keys` list or the default."""
+    for e in env_keys:
+        val = int(os.environ.get(e, -1))
+        if val >= 0:
+            return val
+    return default
+
+
+local_rank = get_int_from_env(["LOCAL_RANK", "MPI_LOCALRANKID"], "0")
+world_size = get_int_from_env(["WORLD_SIZE", "PMI_SIZE"], "1")
+
+deepspeed.init_distributed(get_accelerator().communication_backend_name())
+
+
+def print_rank0(*msg):
+    if local_rank != 0:
+        return
+    print(*msg)
+
+
+# Model loading and instantiating on GPUs
+def get_repo_root(model_name_or_path):
+    if os.path.exists(model_name_or_path):
+        # local path
+        return model_name_or_path
+    # checks if online or not
+    if is_offline_mode():
+        print_rank0("Offline mode: forcing local_files_only=True")
+    # download only on first process
+    allow_patterns = ["*.bin", "*.model", "*.json", "*.txt", "*.py", "*LICENSE"]
+    if local_rank == 0:
+        snapshot_download(
+            model_name_or_path,
+            local_files_only=is_offline_mode(),
+            cache_dir=os.getenv("HF_HOME", None),
+            allow_patterns=allow_patterns,
+            ignore_patterns=["training_args.bin"],
+        )
+
+    dist.barrier()
+
+    return snapshot_download(
+        model_name_or_path,
+        local_files_only=is_offline_mode(),
+        cache_dir=os.getenv("HF_HOME", None),
+        allow_patterns=allow_patterns,
+        ignore_patterns=["training_args.bin"],
+    )
+
+
+def get_checkpoint_files(model_name_or_path):
+    cached_repo_dir = get_repo_root(model_name_or_path)
+
+    # extensions: .bin | .pt
+    # creates a list of paths from all downloaded files in cache dir
+    file_list = [str(entry) for entry in Path(cached_repo_dir).rglob("*.[bp][it][n]") if entry.is_file()]
+    return file_list
+
+
+model_name = "meta-llama/Llama-2-7b-chat-hf"
+tokenizer = AutoTokenizer.from_pretrained(model_name)
+config = AutoConfig.from_pretrained(model_name)
+with deepspeed.OnDevice(dtype=torch.bfloat16, device="meta"):
+    model = AutoModelForCausalLM.from_config(config).to(torch.bfloat16)
+
+model = model.eval()
+model = model.to(memory_format=torch.channels_last)
+
+checkpoints_json = "checkpoints.json"
+
+
+def write_checkpoints_json():
+    checkpoint_files = get_checkpoint_files(model_name)
+    if local_rank == 0:
+        # model.config.model_type.upper()
+        data = {"type": "ds_model", "checkpoints": checkpoint_files, "version": 1.0}
+        json.dump(data, open(checkpoints_json, "w"))
+
+
+repo_root = get_repo_root(model_name)
+
+write_checkpoints_json()
+dist.barrier()
+
+model = deepspeed.init_inference(
```

Comment:
I don't think this makes sense to add this example to optimum-intel as it's mostly unrelated here (only loading the resulting model with `IPEXModelForCausalLM`) what do you think about moving it [here](https://github.com/intel/intel-extension-for-pytorch/tree/main/examples/cpu/inference/python) instead and add a link to it in optimum-intel ?

Reply:
thanks !